src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -230,13 +229,17 @@ static ucs_status_t uct_tcp_sockcm_ep_handle_remote_disconnect(uct_tcp_sockcm_ep
     /* if the server started sending any data that the client received, then
      * it means that the server accepted the client's connection request and
      * created an ep to it. therefore, the server did not reject the request
-     * and if we got here then the status should be UCS_ERR_CONNECTION_RESET.
-     * otherwise, the status is UCS_ERR_REJECTED */
+     * and there was no reject from the network either, and so if we got here
+     * then the status should be UCS_ERR_CONNECTION_RESET.
+     * otherwise, if we got here due to a network reject, we set the status
+     * to UCS_ERR_UNREACHABLE to distinguish between a network reject and
+     * a user's reject (that is done through an explicit message from the server)
+     * which calls the upper layer callback with UCS_ERR_REJECTED. */
     if (ucs_test_all_flags(cep->state, UCT_TCP_SOCKCM_EP_ON_CLIENT |
                                        UCT_TCP_SOCKCM_EP_DATA_SENT) &&
         !(cep->state & (UCT_TCP_SOCKCM_EP_HDR_RECEIVED |
                         UCT_TCP_SOCKCM_EP_DATA_RECEIVED))) {
-        cb_status   = UCS_ERR_REJECTED;
+        cb_status   = UCS_ERR_UNREACHABLE;
```

Comment:
we do we sometimes return UCS_ERR_NOT_CONNECTED and sometimes UCS_ERR_UNREACHABLE for the same case that server rejected the connection on the network level?

Reply:
you mean why we have these cases? https://github.com/openucx/ucx/pull/5460/files#diff-7e368f4ff1888c8f1fa505ebb48d8d94R66

there are different reasons for the reject.

Reply:
no, i mean here it should be UCS_ERR_NOT_CONNECTED instead of UCS_ERR_UNREACHABLE

Reply:
why?

Reply:
because this line happens when the server rejected the connection on the network level

Reply:
how can you tell, on the client side, the reason for the network reject from the server?

Reply:
it doesn't really matter what is the "reason" for network reject. any kind of network reject means the device address could be reached but the connection was dropped/rejected/closed. All of those should be translated to UCS_ERR_NOT_CONNECTED.